File: backend/vortex-admin-pro/src/main/java/com/vortexadmin/service/PreferenceService.java
package com.vortexadmin.service;

import com.vortexadmin.dto.request.PreferenceRequest;
import com.vortexadmin.dto.response.PreferenceResponse;

/**
 * Service contract for managing user-specific UI preferences such as theme, language,
 * and notification settings for the currently authenticated user.
 */
public interface PreferenceService {

    /**
     * Returns the current preferences for the authenticated user.
     * If no preference record exists yet, a default record is returned.
     *
     * @return the user's current preferences
     */
    PreferenceResponse getMyPreferences();

    /**
     * Updates and persists the preferences of the currently authenticated user.
     * Creates the preference record if one does not already exist.
     *
     * @param request the updated preference values
     * @return the saved preference response reflecting the new state
     */
    PreferenceResponse updateMyPreferences(PreferenceRequest request);
}

File: frontend/src/services/preferenceService.js
import api from '../api/axios';

/**
 * Service object for reading and updating the current user's application
 * preferences (e.g. language, theme).
 */
export const preferenceService = {
    /**
     * Fetches the current user's saved preferences.
     * GET /preferences
     * @returns {Promise<import('axios').AxiosResponse>}
     */
    get: () => api.get('/preferences'),

    /**
     * Updates one or more preference fields for the current user.
     * PUT /preferences
     * @param {object} data - Partial preference object to merge (e.g. `{ language: 'th' }`).
     * @returns {Promise<import('axios').AxiosResponse>}
     */
    update: (data) => api.put('/preferences', data),
};
